src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Collections.cs

```diff
@@ -102,10 +104,8 @@ internal sealed partial class ExpressionEval {
         }
 
         public IMember GetValueFromComprehension(Comprehension node) {
-            var oldVariables = CurrentScope.Variables.OfType<Variable>().ToDictionary(k => k.Name, v => v);
-            try {
+            using (Interpreter.LanguageVersion.Is3x() ? OpenScope(Module, node) : Disposable.Empty) {
```

Comment:
Better to hide check inside `OpenScope` by the node type so if we get another place that needs to call `OpenScope` it won't forget the check.

Reply:
great, I'll do that